soc/arm64/nxp_imx/imx8m/mmu_regions.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2020 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <soc.h>
+#include <arch/arm64/arm_mmu.h>
+
+static const struct arm_mmu_region mmu_regions[] = {
+
+	MMU_REGION_FLAT_ENTRY("GIC",
+			      DT_REG_ADDR_BY_IDX(DT_INST(0, arm_gic), 0),
+			      DT_REG_SIZE_BY_IDX(DT_INST(0, arm_gic), 0),
+			      MT_DEVICE_nGnRnE | MT_P_RW_U_RW | MT_NS),
+
+	MMU_REGION_FLAT_ENTRY("GIC",
+			      DT_REG_ADDR_BY_IDX(DT_INST(0, arm_gic), 1),
+			      DT_REG_SIZE_BY_IDX(DT_INST(0, arm_gic), 1),
+			      MT_DEVICE_nGnRnE | MT_P_RW_U_RW | MT_NS),
+
+	MMU_REGION_FLAT_ENTRY("CCM",
+			      DT_REG_ADDR(DT_INST(0, nxp_imx_ccm)),
+			      DT_REG_SIZE(DT_INST(0, nxp_imx_ccm)),
+			      MT_DEVICE_nGnRnE | MT_P_RW_U_RW | MT_NS),
+
+	MMU_REGION_FLAT_ENTRY("UART",
+			      DT_REG_ADDR(DT_INST(0, nxp_imx_iuart)),
+			      DT_REG_SIZE(DT_INST(0, nxp_imx_iuart)),
+			      MT_DEVICE_nGnRnE | MT_P_RW_U_RW | MT_NS),
+
+	MMU_REGION_FLAT_ENTRY("ANA_PLL",
+			      DT_REG_ADDR(DT_INST(0, nxp_imx_ana)),
+			      DT_REG_SIZE(DT_INST(0, nxp_imx_ana)),
+			      MT_DEVICE_nGnRnE | MT_P_RW_U_RW | MT_NS),
+
+	MMU_REGION_FLAT_ENTRY("UART",
+			      DT_REG_ADDR(DT_INST(1, nxp_imx_iuart)),
+			      DT_REG_SIZE(DT_INST(1, nxp_imx_iuart)),
+			      MT_DEVICE_nGnRnE | MT_P_RW_U_RW | MT_NS),
```

Comment:
Hard-coded instance numbers like this are incorrect to use. Please use an alternate mechanism to specify exactly the nodes you want. Read the DT_INST docstring for more details.

Reply:
Could we use a follow up patch to address this? I just follow qemu platform here. @carlocaione  do you have any suggestions?

Reply:
Considering the amount of time this PR has been in an open state (1 year), I think we should merge this and follow up with PRs for fixing issues like this. @mbolivar-nordic what do you think?

Reply:
No, sorry, I don't want to see incorrect usage of the instance API merged.

Reply:
@mbolivar-nordic I am not experts in zephyr, so any suggestions what I should use?